vpn/client.go

```diff
@@ -157,6 +207,10 @@ func (c *Client) SetDeadline(t time.Time) error {
 }
 
 func (c *Client) SetReadDeadline(t time.Time) error {
+	if c.conn == nil {
+		log.Println("CONN IS NIL")
+
+	}
```

Comment:
I think the panic that will ensure is already a good indication of whether the code failed